src/sorting/SortingQuestions.java

```diff
@@ -1,4 +1,43 @@
 package sorting;
 
+import java.util.Arrays;
+
 public class SortingQuestions {
+
+    //1. Implement bubble sort
+    public static int[] bubbleSort(int[] arr){
+        int length = arr.length;    //size of the array
+
+        //looping through all elements in the array arr and for each iteration
+        //it loops through again from first element to the last element and for each iteration
+        //it checks if the adjacent elements are in wrong order and if they are then swapping them to correct the order
+        for (int i = 0; i< length; i++){
+            for (int j = 0; j< length - 1; j++){
```

Comment:
The second loop condition should be j < length - i - 1.  You are currently doing extra comparisons.

After the first pass you know that the last element is the biggest one in the array. After the second pass the second to last element is the second biggest element in the array. This truth follows for every pass in the for loop; Therefore, no need to check if the previous last element is bigger than the current element in every iteration.

Reply:
Wow! I didn't see this coming. I will fix it @neoprez